simtools/db_handler.py

```diff
@@ -25,3 +32,74 @@ def getArrayDB(databaseLocation):
     with open(file, 'r') as stream:
         out = yaml.load(stream, Loader=yaml.FullLoader)
     return out
+
+
+def getModelParameters(telescopeType, version, onlyApplicable=False):
+    '''
+    Read parameters from DB for one specific type (telescopeTYpe, site ...).
+
+    Parameters
+    ----------
+    telescopeType: str
+
+    Returns
+    -------
+    dict containing the parameters
+    '''
+    _telTypeValidated = names.validateName(telescopeType, names.allTelescopeTypeNames)
+    _versionValidated = names.validateName(version, names.allModelVersionNames)
+
+    if getTelescopeSize(_telTypeValidated) == 'MST':
+        # MST-FlashCam or MST-NectarCam
+        _whichTelLabels = [_telTypeValidated, 'MST-optics']
+    elif _telTypeValidated == 'SST':
+        # SST = SST-Camera + SST-Structure
+        _whichTelLabels = ['SST-Camera', 'SST-Structure']
+    else:
+        _whichTelLabels = [_telTypeValidated]
+
+    # Selecting version and applicable (if on)
+    _pars = dict()
+    for _tel in _whichTelLabels:
+        _allPars = collectAllModelParameters(_tel, _versionValidated)
+
+        # If tel is a struture, only applicable pars will be collected, always.
+        # The default ones will be covered by the camera pars.
+        _selectOnlyApplicable = onlyApplicable or _tel in ['MST-optics', 'SST-Structure']
```

Comment:
There is no harm in keeping this optional feature here. For the TelescopeModel we will use onlyApplicable=False, but we may want to turn it on in some other context.
It may actually be important because once we get the parameters, we will lose the information about which parameters are applicable.

Note that selecting only applicable ones for 'MST-optics' and 'SST-Structure' is inevitable by now, because of the issue with the order that we discussed.

Reply:
Sure we can keep it here as well. I do plan to include the "applicable" field in the dictionary you get though (I think), so you won't lose that information.

Reply:
Ok, that will be nice.